projects/arlas-components/src/lib/components/mapgl/model/ArlasMapGL.ts

```diff
@@ -0,0 +1,957 @@
+import { AbstractArlasMapGL, BaseMapGlConfig, BindLayerToEvent, DrawControlsOption, MapEventBinds } from './AbstractArlasMapGL';
+import mapboxgl, {
+  AnyLayer,
+  AnySourceData,
+  Control,
+  FilterOptions,
+  IControl, LngLat, LngLatBoundsLike, LngLatLike, MapboxEvent,
+  MapboxOptions,
+  MapLayerEventType, PointLike
+} from 'mapbox-gl';
+import { MapSource } from './mapSource';
+import { ArlasAnyLayer, MapExtend, paddedBounds } from '../mapgl.component.util';
+import { ControlButton, PitchToggle } from '../mapgl.component.control';
+import { GEOJSON_SOURCE_TYPE, IconConfig, OnMoveResult, VisualisationSetConfig } from '../mapgl.component';
+import { ARLAS_ID, ExternalEvent, FILLSTROKE_LAYER_PREFIX, MapLayers, SCROLLABLE_ARLAS_ID } from './mapLayers';
+import { fromEvent, map } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+import { CdkDragDrop, moveItemInArray } from '@angular/cdk/drag-drop';
+import { ElementIdentifier } from '../../results/utils/results.utils';
+
+export interface ArlasMapGlConfig extends BaseMapGlConfig<MapboxOptions> {
+  mapLayers: MapLayers<AnyLayer>;
+  customEventBind: BindLayerToEvent<keyof MapLayerEventType>[];
+  mapLayersEventBind: {
+    onHover: MapEventBinds<keyof MapLayerEventType>[];
+    emitOnClick: MapEventBinds<keyof MapLayerEventType>[];
+    zoomOnClick: MapEventBinds<keyof MapLayerEventType>[];
+  };
+}
+
+/**
+ *  The aim of this class is to handle all core interaction we have
+ *  with a map provider. And also to handle all new behaviour
+ *  we create from this provider.
+ *
+ * The aim is also to separate the alras map from the angular framework.
+ * The advantage is that in arlas-wui we can use an instance of this class (or not) without worrying about the library used
+ *
+ * it will be instantiated in mapgl and will be responsible for initializing the map and all map behavior.
+ *
+ * Drawing behavior may be handled by another class.
+ *
+ * improvement (open to discussion) :
+ *  - wrap of provider methode could be implemented in another class in middle of
+ *  abstract class.
+ */
+export class ArlasMapGl extends AbstractArlasMapGL {
```

Comment:
The capitalisation of GL should be the same for both (Gl vs GL). Since this map is typed according to mapbox, I think it should be specified in its name to avoid any confusion